source/declarations.tex

```diff
@@ -2038,12 +2038,18 @@
 underlying type as in the original declaration.
 
 \pnum
-If the \grammarterm{enum-key} is followed by a
-\grammarterm{nested-name-specifier}, the \grammarterm{enum-specifier} shall
+If an \grammarterm{enum-head-name} contains a
+\grammarterm{nested-name-specifier}, the latter
+shall not begin with a \grammarterm{decltype-specifier} and
```

Comment:
Was this `decltype` rule for *opaque-enum-declaration*s present before (maybe implied by pre-existing wording)? I see we had such a rule for *enum-head*s.

Reply:
Well, we have (unchanged from before):

If an \grammarterm{opaque-enum-declaration} contains
a \grammarterm{nested-name-specifier},
the declaration shall be an explicit specialization\iref{temp.expl.spec}.

and we have 

11.3 [dcl.meaning]
The nested-name-specifier of a qualified declarator-id shall not begin with a decltype-specifier.

That seems close enough to me. (it seems we don't want redeclarations to use decltype to nominate the entity being redeclared. Which is totally reasonable.)

Reply:
I agree that we don't want to allow `decltype` here, and indeed current implementations don't distinguish between the opaque enum declaration case and the regular enum case, but if this is a normative change (as it appears to be), we should fix that portion of this via the CWG issue process.

Reply:
Fixed by limited the prohibition to the enum-head case.